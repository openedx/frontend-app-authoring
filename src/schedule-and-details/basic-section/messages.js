import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  basicTitle: {
    id: 'course-authoring.schedule.basic.title',
    defaultMessage: 'Basic information',
  },
  basicDescription: {
    id: 'course-authoring.schedule.basic.description',
    defaultMessage: 'The nuts and bolts of this course',
  },
  basicEmailIcon: {
    id: 'course-authoring.schedule.basic.email-icon',
    defaultMessage: 'Invite your students email icon',
  },
  courseOrganization: {
    id: 'course-authoring.schedule.basic.organization',
    defaultMessage: 'Organization',
  },
  courseNumber: {
    id: 'course-authoring.schedule.basic.course-number',
    defaultMessage: 'Course number',
  },
  courseRun: {
    id: 'course-authoring.schedule.basic.course-run',
    defaultMessage: 'Course run',
  },
  basicBannerTitle: {
    id: 'course-authoring.schedule.basic.banner.title',
    defaultMessage: 'Promoting your course with edX',
  },
  basicBannerText: {
    id: 'course-authoring.schedule.basic.banner.text',
    defaultMessage: 'Your course summary page will not be viewable until your course has been announced. To provide content for the page and preview it, follow the instructions provided by your Program Manager. Please note that changes here may take up to a business day to appear on your course summary page.',
  },
  basicPromotionButton: {
    id: 'course-authoring.schedule.basic.promotion.button',
    defaultMessage: 'Invite your students',
  },
});

export default messages;
