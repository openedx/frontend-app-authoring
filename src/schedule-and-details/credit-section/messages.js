import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  creditTitle: {
    id: 'course-authoring.schedule.credit.title',
    defaultMessage: 'Course credit requirements',
  },
  creditDescription: {
    id: 'course-authoring.schedule.credit.description',
    defaultMessage: 'Steps required to earn course credit',
  },
  creditHelp: {
    id: 'course-authoring.schedule.credit.help',
    defaultMessage:
      'A requirement appears in this list when you publish the unit that contains the requirement.',
  },
  creditMinimumGrade: {
    id: 'course-authoring.schedule.credit.minimum-grade',
    defaultMessage: 'Minimum grade',
  },
  creditProctoredExam: {
    id: 'course-authoring.schedule.credit.proctored-exam',
    defaultMessage: 'Successful proctored exam',
  },
  creditVerification: {
    id: 'course-authoring.schedule.credit.verification',
    defaultMessage: 'ID Verification',
  },
  creditNotFound: {
    id: 'course-authoring.schedule.credit.not-found',
    defaultMessage: 'No credit requirements found.',
  },
});

export default messages;
