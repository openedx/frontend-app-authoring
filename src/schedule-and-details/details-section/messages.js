import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  detailsTitle: {
    id: 'course-authoring.schedule-section.details.title',
    defaultMessage: 'Course details',
  },
  detailsDescription: {
    id: 'course-authoring.schedule-section.details.description',
    defaultMessage: 'Provide useful information about your course',
  },
  dropdownLabel: {
    id: 'course-authoring.schedule-section.details.dropdown.label',
    defaultMessage: 'Course language',
  },
  dropdownHelpText: {
    id: 'course-authoring.schedule-section.details.dropdown.help-text',
    defaultMessage: 'Identify the course language here. This is used to assist users find courses that are taught in a specific language. It is also used to localize the \'From:\' field in bulk emails.',
  },
  dropdownEmpty: {
    id: 'course-authoring.schedule-section.details.dropdown.empty',
    defaultMessage: 'Select language',
  },
});

export default messages;
