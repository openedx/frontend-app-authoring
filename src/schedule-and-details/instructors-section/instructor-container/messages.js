import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  instructorNameLabel: {
    id: 'course-authoring.schedule-section.instructor.name.label',
    defaultMessage: 'Name',
  },
  instructorNameHelpText: {
    id: 'course-authoring.schedule-section.instructor.name.help-text',
    defaultMessage: 'Please add the instructor\'s name',
  },
  instructorNameInputPlaceholder: {
    id: 'course-authoring.schedule-section.instructor.name.input.placeholder',
    defaultMessage: 'Instructor name',
  },
  instructorTitleLabel: {
    id: 'course-authoring.schedule-section.instructor.title.label',
    defaultMessage: 'Title',
  },
  instructorTitleHelpText: {
    id: 'course-authoring.schedule-section.instructor.title.help-text',
    defaultMessage: 'Please add the instructor\'s title',
  },
  instructorTitleInputPlaceholder: {
    id: 'course-authoring.schedule-section.instructor.title.input.placeholder',
    defaultMessage: 'Instructor title',
  },
  instructorOrganizationLabel: {
    id: 'course-authoring.schedule-section.instructor.organization.label',
    defaultMessage: 'Organization',
  },
  instructorOrganizationHelpText: {
    id: 'course-authoring.schedule-section.instructor.organization.help-text',
    defaultMessage: 'Please add the institute where the instructor is associated',
  },
  instructorOrganizationInputPlaceholder: {
    id: 'course-authoring.schedule-section.instructor.organization.input.placeholder',
    defaultMessage: 'Instructor organization',
  },
  instructorBioLabel: {
    id: 'course-authoring.schedule-section.instructor.bio.label',
    defaultMessage: 'Biography',
  },
  instructorBioHelpText: {
    id: 'course-authoring.schedule-section.instructor.bio.help-text',
    defaultMessage: 'Please add the instructor\'s biography',
  },
  instructorBioInputPlaceholder: {
    id: 'course-authoring.schedule-section.instructor.bio.input.placeholder',
    defaultMessage: 'Instructor biography',
  },
  instructorPhotoLabel: {
    id: 'course-authoring.schedule-section.instructor.photo.label',
    defaultMessage: 'Photo',
  },
  instructorPhotoHelpText: {
    id: 'course-authoring.schedule-section.instructor.photo.help-text',
    defaultMessage: 'Please add a photo of the instructor (Note: only JPEG or PNG format supported)',
  },
  instructorPhotoInputPlaceholder: {
    id: 'course-authoring.schedule-section.instructor.photo.input.placeholder',
    defaultMessage: 'Instructor photo URL',
  },
  instructorDelete: {
    id: 'course-authoring.schedule-section.instructor.delete',
    defaultMessage: 'Delete',
  },
});

export default messages;
