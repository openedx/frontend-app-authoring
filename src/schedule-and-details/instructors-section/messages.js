import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  instructorsTitle: {
    id: 'course-authoring.schedule-section.instructors.title',
    defaultMessage: 'Instructors',
  },
  instructorsDescription: {
    id: 'course-authoring.schedule-section.instructors.description',
    defaultMessage: 'Add details about the instructors for this course',
  },
  instructorAdd: {
    id: 'course-authoring.schedule-section.instructors.add-instructor',
    defaultMessage: 'Add Instructor',
  },
});

export default messages;
