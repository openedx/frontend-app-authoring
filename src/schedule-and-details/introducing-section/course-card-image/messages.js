import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  courseCardImageLabel: {
    id: 'course-authoring.schedule-section.introducing.course-card-image.label',
    defaultMessage: 'Course card image',
  },
  courseCardUploadButton: {
    id: 'course-authoring.schedule-section.introducing.course-card-image.button',
    defaultMessage: 'Upload course card image',
  },
  courseCardImageHelpText: {
    id: 'course-authoring.schedule-section.introducing.course-card-image.help-text',
    defaultMessage: 'Please provide a valid path and name to your course image (Note: only JPEG or PNG format supported)',
  },
  courseCardImageDescription: {
    id: 'course-authoring.schedule-section.introducing.course-card-image.description',
    defaultMessage: 'Course card image',
  },
  courseCardImageFilesAndUploads: {
    id: 'course-authoring.schedule-section.introducing.course-card-image.files-and-uploads',
    defaultMessage: 'files and uploads',
  },
  courseCardImageInputPlaceholder: {
    id: 'course-authoring.schedule-section.introducing.course-card-image.input.placeholder',
    defaultMessage: 'Your course image URL',
  },
  courseCardImageDropzoneText: {
    id: 'course-authoring.schedule-section.introducing.course-card-image.dropzone-text',
    defaultMessage: 'Drag and drop your course image here or click to upload.',
  },
  courseCardImageEmpty: {
    id: 'course-authoring.schedule-section.introducing.course-card-image.empty',
    defaultMessage: 'Your course currently does not have an image. Please upload one (JPEG or PNG format, and minimum suggested dimensions are 375px wide by 200px tall)',
  },
  courseCardImageIconAlt: {
    id: 'course-authoring.schedule-section.introducing.course-card-image.icon-alt',
    defaultMessage: 'File upload icon',
  },
  courseCardImageBodyFilled: {
    id: 'course-authoring.schedule-section.introducing.course-card-image.manage',
    defaultMessage: 'You can manage this image along with all of your other {hyperlink}',
  },
});

export default messages;
