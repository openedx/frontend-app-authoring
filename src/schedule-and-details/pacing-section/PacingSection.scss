.pacing-section {
  .pgn__form-group {
    p.pgn__form-label {
      font: normal .875rem/1.5rem $font-family-base;
    }

    label.pgn__form-label {
      font: normal $font-weight-bold .75rem/1.25rem $font-family-base;
      color: $text-color-weak;
    }

    .pgn__form-control-description {
      font: normal .75rem/1.5rem $font-family-base;
      color: $black;
    }
  }

  input.pgn__form-radio-input {
    padding: .5rem;
    height: $spacer;
    width: $spacer;
  }

  .pgn__form-control-set {
    padding: $spacer 0 0 2.5rem;
  }
}
