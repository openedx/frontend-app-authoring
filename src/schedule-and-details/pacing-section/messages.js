import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  pacingTitle: {
    id: 'course-authoring.schedule.pacing.title',
    defaultMessage: 'Course pacing',
  },
  pacingDescription: {
    id: 'course-authoring.schedule.pacing.description',
    defaultMessage: 'Set the pacing for this course',
  },
  pacingRestriction: {
    id: 'course-authoring.schedule.pacing.restriction',
    defaultMessage: 'Course pacing cannot be changed once a course has started',
  },
  pacingTypeInstructorLabel: {
    id: 'course-authoring.schedule.pacing.radio.instructor.label',
    defaultMessage: 'Instructor-paced',
  },
  pacingTypeInstructorDescription: {
    id: 'course-authoring.schedule.pacing.radio.instructor.description',
    defaultMessage:
      'Instructor-paced courses progress at the pace that the course author sets. You can configure release dates for course content and due dates for assignments.',
  },
  pacingTypeSelfLabel: {
    id: 'course-authoring.schedule.pacing.radio.self-paced.label',
    defaultMessage: 'Self-paced',
  },
  pacingTypeSelfDescription: {
    id: 'course-authoring.schedule.pacing.radio.self-paced.description',
    defaultMessage: 'Self-paced courses offer suggested due dates for assignments or exams based on the learnerâ€™s enrollment date and the expected course duration. These courses offer learners flexibility to modify the assignment dates as needed.',
  },
});

export default messages;
