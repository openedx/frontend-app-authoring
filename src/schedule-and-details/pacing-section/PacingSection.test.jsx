import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { IntlProvider } from '@edx/frontend-platform/i18n';

import messages from './messages';
import PacingSection from '.';

describe('<PacingSection />', () => {
  const onChangeMock = jest.fn();
  const RootWrapper = (props) => (
    <IntlProvider locale="en">
      <PacingSection {...props} />
    </IntlProvider>
  );

  const props = {
    intl: {},
    selfPaced: false,
    startDate: '2023-06-12',
    onChange: onChangeMock,
  };

  it('renders pacing section successfully', () => {
    const { getByText } = render(<RootWrapper {...props} />);
    expect(getByText(messages.pacingTitle.defaultMessage)).toBeInTheDocument();
    expect(getByText(messages.pacingDescription.defaultMessage)).toBeInTheDocument();
    expect(getByText(messages.pacingRestriction.defaultMessage)).toBeInTheDocument();

    expect(getByText(messages.pacingTypeInstructorLabel.defaultMessage)).toBeInTheDocument();
    expect(getByText(messages.pacingTypeInstructorDescription.defaultMessage)).toBeInTheDocument();

    expect(getByText(messages.pacingTypeSelfLabel.defaultMessage)).toBeInTheDocument();
    expect(getByText(messages.pacingTypeSelfDescription.defaultMessage)).toBeInTheDocument();
  });

  it('shows radio inputs correctly', () => {
    const { getAllByRole } = render(<RootWrapper {...props} />);
    const radioList = getAllByRole('radio');
    expect(radioList[0].checked).toBeTruthy();
    expect(radioList[0].disabled).toBeTruthy();
    expect(radioList[1].checked).toBeFalsy();
    expect(radioList[1].disabled).toBeTruthy();
  });

  it('shows disabled radio inputs correctly', () => {
    const pastDate = '2023-12-31';
    const initialProps = { ...props, startDate: pastDate };
    const { getAllByRole, queryAllByText } = render(
      <RootWrapper {...initialProps} />,
    );
    const radioList = getAllByRole('radio');
    expect(radioList[0].checked).toBeTruthy();
    expect(radioList[0].disabled).toBeFalsy();
    expect(radioList[1].checked).toBeFalsy();
    expect(radioList[1].disabled).toBeFalsy();
    expect(queryAllByText(messages.pacingRestriction.defaultMessage).length).toBe(0);
  });

  it('should call onChange radio input', () => {
    const pastDate = '2023-12-31';
    const initialProps = { ...props, startDate: pastDate };
    const { getAllByRole } = render(<RootWrapper {...initialProps} />);
    const radioList = getAllByRole('radio');
    expect(radioList[1].checked).toBeFalsy();
    fireEvent.click(radioList[1]);
    expect(onChangeMock).toHaveBeenCalled();
  });
});
