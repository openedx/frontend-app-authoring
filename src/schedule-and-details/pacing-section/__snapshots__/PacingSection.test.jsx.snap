// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PacingSection /> matches the snapshots 1`] = `
<section
  className="section-container pacing-section"
>
  <header
    className="section-header"
  >
    <span
      className="lead"
    >
      Course Pacing
    </span>
    <span
      className="x-small text-gray-700"
    >
      Set the pacing for this course
    </span>
  </header>
  <div
    className="pgn__form-group"
  >
    <label
      className="pgn__form-label"
      htmlFor="form-field1"
    >
      Course pacing cannot be changed once a course has started
    </label>
    <div
      className="pgn__form-control-set"
      id="form-field1"
      role="radiogroup"
    >
      <div
        className="pgn__form-radio pgn__form-control-disabled"
      >
        <input
          checked={true}
          className="pgn__form-radio-input"
          disabled={true}
          id="form-field2"
          name="selfPaced"
          onChange={[Function]}
          type="radio"
          value="false"
        />
        <div>
          <label
            className="pgn__form-label"
            htmlFor="form-field2"
          >
            Instructor-Paced
          </label>
          <div
            className="pgn__form-control-description pgn__form-text pgn__form-text-default"
          >
            <div>
              Instructor-paced courses progress at the pace that the course author sets. You can configure release dates for course content and due dates for assignments.
            </div>
          </div>
        </div>
      </div>
      <div
        className="pgn__form-radio pgn__form-control-disabled"
      >
        <input
          checked={false}
          className="pgn__form-radio-input"
          disabled={true}
          id="form-field3"
          name="selfPaced"
          onChange={[Function]}
          type="radio"
          value="true"
        />
        <div>
          <label
            className="pgn__form-label"
            htmlFor="form-field3"
          >
            Self-Paced
          </label>
          <div
            className="pgn__form-control-description pgn__form-text pgn__form-text-default"
          >
            <div>
              Self-paced courses offer suggested due dates for assignments or exams based on the learnerâ€™s enrollment date and the expected course duration. These courses offer learners flexibility to modify the assignment dates as needed.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
